functional/README.md

```diff
@@ -2,52 +2,98 @@
 
-```
-$ ssh-agent
-$ ssh-add fleet/functional/fixtures/id_rsa
-$ echo $SSH_AUTH_SOCK
-/tmp/ssh-kwmtTOsL7978/agent.7978
-```
-2. Ensure the `FLEETD_BIN` and `FLEETCTL_BIN` environment variables point to the respective fleetd and fleetctl binaries that should be used to drive the actual tests.
+### Run tests in Vagrant
+
+The recommended way to run the tests is to use the provided Vagrantfile, which will set up a single CoreOS instance with a one-member etcd cluster (configuration is applied using `user-data` [Cloud-Config][cloud-config] file located in this directory).
+To do so, simply run the following commands on a system with Vagrant installed (see [Vagrant configuration][configure-vagrant] section of this doc)
 
+```sh
+$ git clone https://github.com/coreos/fleet
+$ cd fleet/functional
+$ ./run-in-vagrant
 ```
-$ export FLEETD_BIN=/path/to/fleetd
-$ export FLEETCTL_BIN=/path/to/fleetctl
+
+### Run tests inside other CoreOS platforms (QEMU/BareMetal/libvirt/etc)
+
+It's also possible to run the tests on CoreOS on other platforms. The following commands should be run *inside* the CoreOS instance.
+
+```sh
+$ git clone https://github.com/coreos/fleet
 ```
 
-3. Make sure etcd is running on the host system.
+If you didn't configure etcd2 daemon yet, just run this script:
 
+```sh
+$ sudo fleet/functional/start_etcd
 ```
-$ systemctl start etcd
+
+It will configure and start a one-member etcd cluster.
+
+Then run the functional tests (script will download and unpack golang into home directory):
+
+```sh
+$ sudo fleet/functional/test
 ```
 
-Then the tests can be run with:
+## Configure host environment to run Vagrant
 
+### Debian/Ubuntu
+
+#### Install Vagrant
+
+```sh
+sudo apt-get install -y git nfs-kernel-server
+wget https://releases.hashicorp.com/vagrant/1.8.1/vagrant_1.8.1_x86_64.deb
+sudo dpkg -i vagrant_1.8.1_x86_64.deb
 ```
-# go test github.com/coreos/fleet/functional
+
+#### Install VirtualBox
+
+```sh
+echo "deb http://download.virtualbox.org/virtualbox/debian $(lsb_release -sc) contrib" | sudo tee /etc/apt/sources.list.d/virtualbox.list
+wget -q https://www.virtualbox.org/download/oracle_vbox.asc -O- | sudo apt-key add -
+sudo apt-get update
+sudo apt-get install -y build-essential dkms
+sudo apt-get install -y VirtualBox-5.0
+#Previous VirtualBox (if you have problems with nested virtualization, more info here: https://www.virtualbox.org/ticket/14965)
+#sudo apt-get install -y VirtualBox-4.3
 ```
 
-Since the tests utilize `systemd-nspawn`, this needs to be invoked as sudo/root.
+### CentOS/Fedora
 
-An example test session using coreos-vagrant follows. This assumes that go is available in `/home/core/go` and the fleet repository in `/home/core/fleet` on the target machine (the easiest way to achieve this is to use shared folders).
+**NOTE**: NFS and Vagrant doesn't work out of the box on CentOS 6.x
```

Comment:
Will add `**NOTE**: NFS and Vagrant doesn't work out of the box on CentOS 6.x, so it is recommended to use CentOS 7.x`